source/AI.cpp

```diff
@@ -3292,20 +3292,26 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 
-			double closest = numeric_limits<double>::infinity();
+			double best = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
 			bool foundAnything = false;
+			bool distancePriority = Preferences::Has("Board target");
+			double agility = ship.Acceleration() * ship.TurnRate();
 			for(const shared_ptr<Ship> &other : ships)
 				if(CanBoard(ship, *other))
 				{
 					if(shift && !other->IsYours())
 						continue;
 
 					bool isEnemy = other->GetGovernment()->IsEnemy(ship.GetGovernment());
-					double d = other->Position().Distance(ship.Position());
-					if((isEnemy && !foundEnemy) || (d < closest && isEnemy == foundEnemy))
+					double distance = other->Position().DistanceSquared(ship.Position());
+					double cost = Has(ship, other, ShipEvent::SCAN_OUTFITS) ? 
+						other->Cost() : (other->ChassisCost() * 2.);
+					double b = distancePriority ? distance : 
```

Comment:
If my ship is on top of another ship (for example because I was boarding while cloaked), could that then lead to a division by zero at this point?